module-gtnhmixins/src/main/java/com/gtnewhorizon/gtnhmixins/builders/TargetModBuilder.java

```diff
@@ -0,0 +1,176 @@
+     * {@link TargetModBuilder#setTargetClass}. Requires you to specify a class to target via {@link TargetModBuilder#setTargetClass}.
+     */
+    public TargetModBuilder setClassNodeTest(Predicate<ClassNode> classNodeTest) {
+        this.classNodeTest = classNodeTest;
+        return this;
+    }
+
+    /**
+     * A conditional check that will test the contents of the @Mod annotation of the class specified in
+     * {@link TargetModBuilder#setTargetClass}. Requires you to specify a class to target via {@link TargetModBuilder#setTargetClass}.
+     * Tests if the modId of the target class is equals to the modId provided.
+     */
+    public TargetModBuilder testModID(String modId) {
+        this.classNodeTest = Predicates.testModAnnotation(Predicates.equals(modId), null, null);
+        return this;
+    }
+
+    /**
+     * A conditional check that will test the contents of the @Mod annotation of the class specified in
+     * {@link TargetModBuilder#setTargetClass}. Requires you to specify a class to target via {@link TargetModBuilder#setTargetClass}.
+     * Tests if the modId and modVersion of the target class are equal to the modId and modVersion provided.
+     */
+    public TargetModBuilder testModVersion(String modId, String modVersion) {
+        this.classNodeTest = Predicates.testModAnnotation(Predicates.equals(modId), null, Predicates.equals(modVersion));
+        return this;
+    }
+
+    /**
+     * A conditional check that will test the contents of the @Mod annotation of the class specified in
+     * {@link TargetModBuilder#setTargetClass}. Requires you to specify a class to target via {@link TargetModBuilder#setTargetClass}.
+     * Tests if the modId of the target class is equals to the modId provided and tests the modVersion with the provided test.
+     */
+    public TargetModBuilder testModVersion(String modId, Predicate<String> modVersionTest) {
+        this.classNodeTest = Predicates.testModAnnotation(Predicates.equals(modId), null, modVersionTest);
+        return this;
+    }
+
+    /**
+     * A conditional check that will test the contents of the @Mod annotation of the class specified in
+     * {@link TargetModBuilder#setTargetClass}. Requires you to specify a class to target via {@link TargetModBuilder#setTargetClass}.
+     * Some arguments can be null but not all at once.
+     */
+    public TargetModBuilder testModAnnotation(Predicate<String> modIdTest, Predicate<String> modNameTest, Predicate<String> modVersionTest) {
+        this.classNodeTest = Predicates.testModAnnotation(modIdTest, modNameTest, modVersionTest);
+        return this;
+    }
+
+    ///**
+    // * A conditional check that will test the name of the jar files. !!! This mod identification method should only be
+    // * used as a last resort if you cannot identify your targeted mod with the other methods !!!
+    // */
+    //public TargetModBuilder setJarNameTest(Predicate<String> jarNameTest) {
```

Comment:
Forgotten dead code?

Reply:
That was for FalsePattern to implement jar name matching but idk if that's still needed with the Mod annotation test